### ISSUE TITLE
Forge 1.17.1 can't be installed. | Could not find or load main class cpw.mods.bootstraplauncher.BootstrapLauncher

### ISSUE THREAD

**Issue by SwanX1** [gorilla-devs/GDLauncher#998]
**Describe the bug**
Forge 1.17.1 37.0.1 can't be installed.

**To Reproduce**
Just try to make an instance of Minecraft 1.17.1 using Forge 37.0.1, it will be stuck at "100% Downloading game files"

**Expected behavior**
The instance should install (kinda obvious)

**Possible solutions**
Launching the instance after closing and reopening the launcher throws the following error:
`Error: Could not find or load main class cpw.mods.bootstraplauncher.BootstrapLauncher`
After doing some deep diving, it seems that `libraries/cpw/mods/bootstraplauncher/0.1.15/bootstraplauncher-0.1.15.jar` hasn't been downloaded, that is because it doesn't exist. only 0.1.14 exists. See:
https://files.minecraftforge.net/cpw/mods/bootstraplauncher/
![image](https://user-images.githubusercontent.com/36593309/126829794-bd3c56e0-ec67-4c2a-94fb-00cb1e67b563.png)

A possible solution would be trying to download 0.1.14 instead of 0.1.15

**Operating System:**
 - OS: Linux (Ubuntu 20.04) (using appimage)
 - Version 1.1.11

**Comment by hypherionmc** [comment]
Same issue on Windows

**Comment by ramonGonzEdu** [comment]
Copied my comment from #1008 

If you download the latest forge installer JSON, unzip it (might need to rename), the install_profile.json file has the following line:
```json
{
    "_comment_": [
        "Please do not automate the download and installation of Forge.",
        "Our efforts are supported by ads from the download page.",
        "If you MUST automate this, please consider supporting the project through https://www.patreon.com/LexManos/"
    ],
    "spec": 1,
    "profile": "forge",
    "version": "1.17.1-forge-37.0.12",
    "path": null,
    "minecraft": "1.17.1"
    
}
```

After the minecraft version, there is a large list of deps. the path property is set to null which is not handled on GDLauncher and causes an error on the function `mavenToArray` since the first line calls `s.split` without nullchecking any function parameters. This definitely seems to be an issue with GDLauncher since JSON from a 3rd party is being blindly read without any null checks or sanity checks to prevent this.

Edit: The line in the json that breaks GDL is the `"path": null,`

**Comment by Shad0wyDr3ams** [comment]
Copied my comment from #1008

this should be fixable via adding this to path line
 "path": "net.minecraftforge:forge:1.17.1-37.0.12"

 the reason i say this is because in every other version of forge before 1.17-1.17.1, that is how the path is named with the exception of the numbers being different per version.

**Comment by Arrowpuppet** [comment]
> Copied my comment from #1008
> 
> this should be fixable via adding this to path line
> "path": "net.minecraftforge:forge:1.17.1-37.0.12"
> 
> the reason i say this is because in every other version of forge before 1.17-1.17.1, that is how the path is named with the exception of the numbers being different per version.

I've got this line fixed in the installer but how do I get GDL to actually install it? The only option to install custom versions is for custom zip files and that doesn't seem to work with this installer

**Comment by Shad0wyDr3ams** [comment]
> > Copied my comment from #1008
> > 
> > this should be fixable via adding this to path line
> > "path": "net.minecraftforge:forge:1.17.1-37.0.12"
> > 
> > the reason i say this is because in every other version of forge before 1.17-1.17.1, that is how the path is named with the exception of the numbers being different per version.
> 
> I've got this line fixed in the installer but how do I get GDL to actually install it? The only option to install custom versions is for custom zip files and that doesn't seem to work with this installer


To my knowledge GDLauncher has no way of allowing custom forge installs... this being said forge itself needs to implement the fix or gd launcher needs to implement a bypass in their launcher so it doesnt read the path line. Or give people the option to do custom forge installs.

**Comment by hypherionmc** [comment]
Fixed in release 1.1.12+

**Comment by jclendineng** [comment]
Same issue here on windows 11 arm, 1.16.5 works while 1.18.1 does not, fails with the mentioned error.
